doc/whats-new.rst

```diff
@@ -74,6 +74,7 @@ Deprecations
 
 Bug fixes
 ~~~~~~~~~
+- Ensure times after 2262 are correctly decoded and don't silently overflow. By `Zeb Nicholls <https://github.com/znicholls>`_.
```

Comment:
Awesome, thanks @znicholls -- I want to see if we can make this note a little more specific.  Does this look ok to you?
```suggestion
- Ensure standard calendar times encoded with large values (i.e. greater than approximately 292 years), can be decoded correctly without silently overflowing (:pull:`5050`).  This was a regression in xarray 0.17.0.  By `Zeb Nicholls <https://github.com/znicholls>`_.
```

Reply:
Looks great